pkg/autoscaler/autoscaler.go

```diff
@@ -291,6 +296,138 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+			workloadArch := framework.Amd64
+			expectedReplicas := int32(1)
+			var expectedScaledMachineSet *machinev1.MachineSet
+			var machineSets = make([]*machinev1.MachineSet, 0)
+
+			for _, machineSetParams := range machineSetParamsList {
+				machineSetParams.Labels[targetedNodeLabel] = ""
+				By(fmt.Sprintf("Deploying the MachineSet with 0 replicas (architecture: %s)",
+					machineSetParams.Labels[framework.ArchLabel]))
+				machineSet, err := framework.CreateMachineSet(client, machineSetParams)
+				Expect(err).ToNot(HaveOccurred(), "Failed to create MachineSet with 0 replicas")
+				machineSets = append(machineSets, machineSet)
+				cleanupObjects[machineSet.GetName()] = machineSet
+
+				framework.WaitForMachineSet(ctx, client, machineSet.GetName())
+
+				By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s/%s - min:%v, max:%v",
+					machineSet.GetNamespace(), machineSet.GetName(), 0, expectedReplicas))
+				asr := machineAutoscalerResource(machineSet, 0, expectedReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed(), fmt.Sprintf("Failed to create MachineAutoscaler with min 0/max %v replicas", expectedReplicas))
+				cleanupObjects[asr.GetName()] = asr
+				arch := machineSetParams.Labels[framework.ArchLabel]
+				if arch != framework.Amd64 {
+					workloadArch = arch
+					expectedScaledMachineSet = machineSet
+				}
+			}
+			if workloadArch == "" {
+				// If workloadArch is empty, it means that exactly one MachineSet is in the list, and it is amd64.
+				workloadArch = framework.Amd64
+				expectedScaledMachineSet = machineSets[0]
+			}
+			uniqueJobName := fmt.Sprintf("%s-scale-from-zero", workloadJobName)
+			By(fmt.Sprintf("Creating scale-out workload %s: jobs: %v, memory: %s", uniqueJobName,
+				expectedReplicas+1, workloadMemRequest.String()))
+			// Executing one additional replica to ensure that the workload does not triggers other scale-out events with
+			// other MachineSets not having the requested architecture.
+			workload := framework.NewWorkLoad(expectedReplicas+1, workloadMemRequest, uniqueJobName, autoscalingTestLabel, "",
+				corev1.NodeSelectorRequirement{
+					Key:      targetedNodeLabel,
+					Operator: corev1.NodeSelectorOpExists,
+				},
+				corev1.NodeSelectorRequirement{
+					Key:      "kubernetes.io/arch",
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{workloadArch},
+				})
+			// Setting the command to sleep 30 seconds to ensure that the workload can be completed and the additional
+			// replica (job completion) is run on a node with the requested architecture before the scale-in event is triggered.
+			workload.Spec.Template.Spec.Containers[0].Command = []string{"sleep", "30"}
+
+			cleanupObjects[workload.GetName()] = workload
+			Expect(client.Create(ctx, workload)).Should(Succeed(), "Failed to create scale-out workload %s", workloadJobName)
+
+			Consistently(func() bool {
```

Comment:
is this `Consistently` clause to ensure that we don't scale the wrong machineset?

Reply:
yes, it is.